### ISSUE TITLE
Json file doesn't be saved with the new value 

### ISSUE THREAD

**Issue by pepson84** [bblanchon/ArduinoJson#1825]
**Description**
hello there. 
what i would like to do is to update in the main loop the values of some variables contained in a json file. but it doesn't work

**Troubleshooter's report**
1. The issue happens at run time
2. The issue concerns deserialization
3. <code>deserializeJson()</code> returns <code>IncompleteInput</code>
4. Input comes from a pointer
5. Increasing the buffer size doesn't solve the issue
6. Reading from stream doesn't fix the issue

**Environment**
* Microcontroller: ESP8266
* Core/Framework: ESP8266 core 3.0.2 using Arduino  IDE 1.8.19
* IDE: Arduino IDE 1.8.19

**Reproduction code**
```c++

void loop() {
  
BlynkEdgent.run();

  timer.run();
 
  if(conta=1)
{ 

enerday=4.5;

}
conta++;
enertot+=enerday*2;

if(conta%2==0){

 doc["enerday"]=enerday;
 doc["enertot"]=enertot;
 doc["gennaio"]=enertot*2;
 doc["autoc_gennaio"]=enertot*3;
 doc["enerim_gennaio"]=enertot*4;
 delay(100); 
serializeJson(doc, eepromStream);
serializeJsonPretty(doc, Serial);
EEPROM.commit();

}

}
```

**Remarks**
it seems that in some address of the heap it can save right the wifi settings, but not the json "doc" i want to use to store and upgrade the variables. 

the same complete script, using EEPROM library works fine. 

thanks for your support.

**Comment by bblanchon** [comment]
Hi @pepson84,

There are too many moving parts in this program.
Please remove every irrelevant line and keep it under 20 lines.

In your message, please also include a comparison of what works and what doesn't.

Best regards,
Benoit

**Comment by pepson84** [comment]
> Hi @pepson84,
> 
> There are too many moving parts in this program. Please remove every irrelevant line and keep it under 20 lines.
> 
> In your message, please also include a comparison of what works and what doesn't.
> 
> Best regards, Benoit

Hello @bblanchon,
thanks, i removed the irrelevant line. 
thanks

**Comment by bblanchon** [comment]
I'm puzzled because the Troubleshooter report says, "deserializeJson() returns IncompleteInput", but the program doesn't contain any call to `deserializeJson()`.

Anyway, the only thing I noticed on your program is the missing definition of `eepromStream`.
You must declare `eepromStream` inside the `loop()` function, preferably on the line above `serializeJson()`; otherwise, it will append instead of overriding.
In other words, the `EepromStream` doesn't automatically rewind, so you need to create a new one.
[*Be kind rewind*](https://youtu.be/_NRoUAe3NW0) 😉

**Comment by pepson84** [comment]
thanks for your support @bblanchon; the deserializeJson() was in the Setup, as also the definition of eepromStream.
i moved the eeproStream decleration in the loop() and it works, but the i come back to into another issue, that i ha founded during my previous test: after some reset/reboot of the esp8266 it lost the wifi settings (note i'm using the BlynkEdgent function). 


```
void setup()
{
  Serial.begin(115200);
  delay(100);
mySerial.begin(115200);
BlynkEdgent.begin();
   EEPROM.begin(4096);
}
void loop() {
EepromStream eepromStream(900, 4000);
deserializeJson(doc, eepromStream); 
   Serial.println("doc nel loop");
serializeJsonPretty(doc, Serial);
BlynkEdgent.run();

  if(conta==1)
{ 
 enerday=4.5;
enertot=doc["enertot"];
}

conta++;
enerday+=enerday;
enertot+=enerday*2;



if(conta%2==0){
  Serial.println("conta: "); 
Serial.println(conta);

 //deserializeJson(doc, eepromStream);
 doc["enerday"]=enerday;
 doc["enertot"]=enertot;
 doc["gennaio"]=enertot*2;
 doc["autoc_gennaio"]=enertot*3;
 doc["enerim_gennaio"]=enertot*4;
 delay(100); 
serializeJson(doc, eepromStream);
serializeJsonPretty(doc, Serial);
EEPROM.commit();
delay(2000); 
  Serial.println(" eeprom in the LOOP");
 Serial.println( EEPROM.length());
}

}
```


this is the output

```
22:14:50.511 -> doc nel loop
22:14:50.558 -> {
22:14:50.558 ->   "enerday": 9,
22:14:50.558 ->   "enertot": 2232,
22:14:50.558 ->   "gennaio": 4464,
22:14:50.558 ->   "autoc_gennaio": 6696,
22:14:50.558 ->   "enerim_gennaio": 8928
22:14:50.558 -> }conta: 
22:14:50.558 -> 2
22:14:50.558 -> {
22:14:50.558 ->   "enerday": 9,
22:14:50.558 ->   "enertot": 2250,
22:14:50.558 ->   "gennaio": 4500,
22:14:50.558 ->   "autoc_gennaio": 6750,
22:14:50.558 ->   "enerim_gennaio": 9000
22:14:50.558 -> }{
```

here i do reboot

```
[390] Using default config.
22:14:51.216 -> [390] 
22:14:51.216 ->     ___  __          __
22:14:51.216 ->    / _ )/ /_ _____  / /__
22:14:51.216 ->   / _  / / // / _ \/  '_/
22:14:51.216 ->  /____/_/\_, /_//_/_/\_\
22:14:51.216 ->         /___/ v1.1.0 on ESP8266
22:14:51.216 -> 
22:14:51.216 ->  #StandWithUkraine    https://bit.ly/swua
22:14:51.216 -> 
22:14:51.216 -> 
22:14:51.216 -> [398] --------------------------
22:14:51.216 -> [401] Product:  Meter
22:14:51.216 -> [403] Firmware: 10.0.0 (build Nov 21 2022 22:12:57)
22:14:53.253 -> [408] Device:   ESP8266 @ 80MHz
22:14:53.253 -> [411] MAC:      xx:Xx:xx:XX:xx:xx
22:14:53.253 -> [414] Flash:    4096K
22:14:53.253 -> [416] ESP core: 3.0.2
22:14:53.253 -> [418] ESP SDK:  2.2.2-dev(38a443e)
22:14:53.253 -> [421] Boot Ver: 6
22:14:53.253 -> [422] Boot Mode:1
22:14:53.253 -> [424] FW info:  509984/1585152, MD5:16b22e3ccfaf75a0b247e3d02ba1d202
22:14:53.253 -> [650] Free mem: 23984
22:14:53.253 -> [650] --------------------------
22:14:53.253 -> 
22:14:53.253 -> >[650] INIT => WAIT_CONFIG
22:14:53.253 ->  eeprom ispnibile setup
22:14:53.253 -> 4096
22:14:53.253 -> doc nel loop
22:14:53.253 -> null[1262] AP SSID: Blynk Meter-XXXX
22:14:53.253 -> [1262] AP IP:   192.168.4.1
22:14:53.253 -> [1263] AP URL:  blynk.setup

```

thanks for your support

**Comment by bblanchon** [comment]
I don't know why the Blynk configuration is lost; that's probably a question you should address with Blynk's support team.

BTW, there is no hardware EEPROM on the ESP8266.
Instead, the EEPROM library emulates the behavior using a flash sector (hence the additional `commit()` function).
I suggest you try using LittleFS or SPIFFS to store your JSON document.
Hopefully, this will solve all your issues.

**Comment by pepson84** [comment]
i'll let you know; thank you so much for your help.
